cmd/consumer/main.go

```diff
@@ -26,14 +27,15 @@ func New() *SampleRecordProcessor {
 	}
 }
 
-func (srp *SampleRecordProcessor) Initialize(shardID string) error {
+func (srp *SampleRecordProcessor) Initialize(shardID string, checkpointer *kcl.Checkpointer) error {
 	srp.lastCheckpoint = time.Now()
+	srp.checkpointer = checkpointer
 	return nil
 }
 
-func (srp *SampleRecordProcessor) checkpoint(checkpointer kcl.Checkpointer, sequenceNumber *string, subSequenceNumber *int) {
-	for n := -1; n < srp.checkpointRetries; n++ {
-		err := checkpointer.Checkpoint(sequenceNumber, subSequenceNumber)
+func (srp *SampleRecordProcessor) checkpoint(sequenceNumber *string, subSequenceNumber *int) {
```

Comment:
I think we could also move this out of each record processor and into a shared place (the `kcl` package?)

either

```go
func checkpoint(sequenceNumber *string, subSequenceNumber *int, retries int, sleepDuration time.Duration) {
```

or have set retry behavior, not configurable by consumer

```go
func checkpoint(sequenceNumber *string, subSequenceNumber *int) {
```